browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -223,28 +263,65 @@ void BraveNewTabPageHandler::OnSavedCustomImage(const base::FilePath& path) {
     return;
   }
 
-#if defined(OS_WIN)
-  auto file_name = base::SysWideToUTF8(path.BaseName().value());
-#else
-  auto file_name = std::string(path.BaseName().value().c_str());
-#endif
+  if (brave_new_tab_page::mojom::kMaxCustomImageBackgrounds -
+          NTPBackgroundPrefs(profile_->GetPrefs())
+              .GetCustomImageList()
+              .size() <=
+      0) {
```

Comment:
Curious this `if` clause can be `true`?
As `MultiFilesSelected()` controls the total count, I think we can replace with `DCHECK`?

Reply:
Yeah, I've experienced this. When a user try uploading an image with high resolution, like > 1mb, it talks a few secs. If the user upload another image at that moment, this can happen.

Reply:
Got it!